creator/ingest_requests/models.py

```diff
@@ -1,16 +1,160 @@
+from django.contrib.auth import get_user_model
+from django.db.models.signals import pre_save
+from django.core.exceptions import ValidationError, FieldDoesNotExist
 from django.db import models
 
+from creator.files.models import Version
+from creator.jobs.models import Job
+
+DELIMITER = "-"
+NAME_PREFIX = "INGEST_REQUEST"
+
+User = get_user_model()
+
 
 class IngestRequest(models.Model):
     """
-    Plug the model in here
+    Request to ingest file(s) into a target data service
     """
 
     class Meta:
         permissions = [
             ("list_all_ingestrequest", "Show all ingest_requests"),
         ]
 
+    name = models.CharField(
```

Comment:
Is this being used as a primary key, or a way to constrain uniqueness so that you may only have one ingest run for a particular set of versions?

Reply:
Constructing this field seems to create some complications in creating the model. Could it instead be a computed property?

```
@property
def name(self):
    return ''.join(v.name for v in self.versions.all())
```